### PR TITLE
Remove shard prefix from upload location

### PR DESCRIPTION
**Pull request by Dread1982**
# TL;DR
~~Reorder prefixes for uploads in bucket. StoragePrefix should come before shard prefix.~~
Remove shard prefix from upload path.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
~~Paths for uploads now follow the same convention like e.g. for data, instead of writing shard-prefixes directly in the root folder of the bucket.~~
Paths for uploads now don't contain a shard prefix. See discussion below for details.

Before:
`<bucket-name>/<shard-prefix>/<storage-prefix>`
e.g. `my-bucket/xy/upload/some-domain/...`

Now:
~~`<bucket-name>/<storage-prefix>/<shard-prefix>`~~
~~e.g. `my-bucket/upload/xy/some-domain/...`~~
`<bucket-name>/<storage-prefix>`
e.g. `my-bucket/upload/some-domain/...`

## Tracking Issue

fixes https://github.com/flyteorg/flyte/issues/2545

## Follow-up issue
_NA_

### CODE REVIEW

File: dataproxy/service.go [hunk 1]
```diff
@@ -140,9 +140,6 @@ func (s Service) validateCreateDownloadLocationRequest(req *service.CreateDownlo
 func createShardedStorageLocation(ctx context.Context, shardSelector ioutils.ShardSelector, store *storage.DataStore,
 	cfg config.DataProxyUploadConfig, keyParts ...string) (storage.DataReference, error) {
 	keySuffixArr := make([]string, 0, 4)
-	if len(cfg.StoragePrefix) > 0 {
```
Comment (EngHabu) [inline dataproxy/service.go:143]: Since the only thing we now put in keySuffix is keyParts, I think we can get rid of the former completely and avoid the extra initialization

Comment (Dread1982) [reply dataproxy/service.go:143]: Good point. I adjusted it.

**Comment by sbrunk** [comment]
Hey @EngHabu could you trigger the CI again? Looks like I don't have permissions for the flyteadmin repo yet to do that.

**Review by wild-endeavor — APPROVED** [review]

**Review by EngHabu — CHANGES_REQUESTED** [review]
Can we plz hold off I'm checking on s3 recommendation for maximizing read/write speed

**Comment by wild-endeavor** [comment]
@Dread1982 sorry for the back and forth, but could you update the PR to just remove the shard prefix entirely?  The addition of the shard prefix at this point seems a bit like premature optimization, esp given that the prefix already has project and domain, the suffix can be specified in the request, and the S3 doc seems to imply that it's possible to shard, if needed, across the entire length of the prefix and not just the first couple characters.

See https://docs.aws.amazon.com/AmazonS3/latest/userguide/optimizing-performance.html for more.

**Comment by Dread1982** [comment]
@wild-endeavor @EngHabu I updated the PR as suggested. Please have a look.

**Review by wild-endeavor — APPROVED** [review]

**Review by EngHabu — APPROVED** [review]
Thank you for updating the PR!
